go-controller/pkg/libovsdb/ops/chassis.go

```diff
@@ -151,7 +151,8 @@ func CreateOrUpdateChassis(sbClient libovsdbclient.Client, chassis *sbdb.Chassis
 		},
 		{
 			Model:            chassis,
-			OnModelMutations: []interface{}{&chassis.OtherConfig, &chassis.Encaps},
+			OnModelMutations: []interface{}{&chassis.OtherConfig},
```

Comment:
This wont work because the logic is to apply only one of these. It prefers updates over mutations:

```
	doWhenFound := func(model interface{}, opModel *operationModel) ([]ovsdb.Operation, error) {
		// nil represents onModelUpdatesNone
		if opModel.OnModelUpdates != nil {
			return m.update(model, opModel)
		} else if opModel.OnModelMutations != nil {
			return m.mutate(model, opModel, ovsdb.MutateOperationInsert)
		}
```

https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/libovsdb/ops/model_client.go#L222

I'm not sure why the code is this way and why we coudlnt do both and return both ops. @jcaamano ?

The other option is you do 2 different txns to update encaps and other config, but those transactions have to be transacted separately. Otherwise if the model didnt exist, libovsdb will try to make 2 creates for these models, and that will conflict during txn time.

Reply:
It worked in my scenario. I saw the duplicate encap-ip disappearing.

Reply:
yeah it will work for encap update, but any mutations will not happen

Reply:
wait, how about doing this instead:

```
// CreateOrUpdateChassis creates or updates the chassis record along with the encap record
func CreateOrUpdateChassis(sbClient libovsdbclient.Client, chassis *sbdb.Chassis, encap *sbdb.Encap) error {
	m := newModelClient(sbClient)
	opModels := []operationModel{
		{
			Model: encap,
			DoAfter: func() {
				chassis.Encaps = []string{encap.UUID}
			},
			OnModelUpdates: onModelUpdatesAllNonDefault(),
			ErrNotFound:    false,
			BulkOp:         false,
		},
		{
			Model:            chassis,
			OnModelUpdates:   []interface{}{&chassis.Encaps},
			ErrNotFound:      false,
			BulkOp:           false,
		},
		{
			Model:            chassis,
			OnModelMutations: []interface{}{&chassis.OtherConfig},
			ErrNotFound:      false,
			BulkOp:           false,
		},
	}

	if _, err := m.CreateOrUpdate(opModels...); err != nil {
		return err
	}

	return nil
}
```

Reply:
> I'm not sure why the code is this way and why we coudlnt do both and return both ops. @jcaamano ?

We could. We just wouldn't be able to define a good behavior if the same column is specified as update and mutation at the same time. But as that doesn't make any sense, we can just say that's undefined behavior.

Reply:
sounds good to me. @wizhaoredhat do you want to make that change in this PR or would prefer I do it in another PR and then you rebase?

Reply:
@trozet You mean @flavio-fernandes 's changes? I have tested his changes and it works on my end. I could update this PR with his changes.

Reply:
No, those changes will work, but at the cost of an extra transaction. Jaime and I are in agreement that we can change the library to generate both mutate and update operations for a model if both are requested. If you want I can push a PR with that change or you can do it as another commit here. Just let me know.

Reply:
@trozet I think it would be better for you to push the PR with the change.

Reply:
@trozet I just realized that the reason why the tests are failing is probably because we need the change in library. Maybe you can take this PR over? I don't mind.
```
2023-10-16T22:14:23.8979356Z [91m[1m• Failure [0.325 seconds][0m
2023-10-16T22:14:23.8979841Z Zone Interconnect Chassis Operations
2023-10-16T22:14:23.8981133Z [90m/home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/ovn/zone_interconnect/chassis_handler_test.go:21[0m
2023-10-16T22:14:23.8982249Z   [91m[1mSync nodes [It][0m
2023-10-16T22:14:23.8983440Z   [90m/home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/ovn/zone_interconnect/chassis_handler_test.go:342[0m
2023-10-16T22:14:23.8984428Z 
2023-10-16T22:14:23.8984654Z   [91mExpected
2023-10-16T22:14:23.8985055Z       <map[string]string | len:0>: nil
2023-10-16T22:14:23.8985495Z   to have {key: value}
2023-10-16T22:14:23.8986036Z       <map[interface {}]interface {} | len:1>: {
2023-10-16T22:14:23.8986786Z           <string>"is-remote": <string>"false",
2023-10-16T22:14:23.8987344Z       }[0m
2023-10-16T22:14:23.8987576Z 
2023-10-16T22:14:23.8988469Z   /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/ovn/zone_interconnect/chassis_handler_test.go:363
```

Reply:
Sure I can just push to this PR. Will add a commit.

Reply:
oh wait I see Flavio's comment

Reply:
heh, sorry, Tim. Mid-air collision. ;)